### PR TITLE
Add gymnasium support for TD3

### PR DESCRIPTION
**Pull request by pseudo-rnd-thoughts**
## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] New algorithm
- [ ] Documentation

## Checklist:

- [x] I've read the [CONTRIBUTION](https://docs.cleanrl.dev/contribution/) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [x] I have updated the tests accordingly (if applicable).
- [ ] I have updated the documentation and previewed the changes via `mkdocs serve`.
    - [ ] I have explained note-worthy implementation details.
    - [ ] I have explained the logged metrics.
    - [ ] I have added links to the original paper and related papers.

If you need to run benchmark experiments for a performance-impacting changes:

- [ ] I have contacted @vwxyzjn to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark).
- [ ] I have used the [benchmark utility](/get-started/benchmark-utility/) to submit the tracked experiments to the [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) W&B project, optionally with `--capture-video`.
- [ ] I have performed RLops with `python -m openrlbenchmark.rlops`.
    - For new feature or bug fix:
        - [ ] I have used the RLops utility to understand the performance impact of the changes and confirmed there is no regression.
    - For new algorithm:
        - [ ] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [ ] I have added the learning curves generated by the `python -m openrlbenchmark.rlops` utility to the documentation.
    - [ ] I have added links to the tracked experiments in W&B, generated by `python -m openrlbenchmark.rlops ....your_args... --report`,  to the documentation.

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Hi @pseudo-rnd-thoughts thanks for the PR! Could you do a filediff between td3 variants and ddpg variants to minimize the lines of code differences? You can select two files in vscode, right click them, and select "compare selected", which should produce something like below. 

image

**Comment by pseudo-rnd-thoughts** [comment]
> Hi @pseudo-rnd-thoughts thanks for the PR! Could you do a filediff between td3 variants and ddpg variants to minimize the lines of code differences? You can select two files in vscode, right click them, and select "compare selected", which should produce something like below.

Done in fed8aaf09c880e7e8df9a517a850c51cede99a29
The differences between DDPG and TD3 should be solely the heuristic improvements. The clipping part is relatively confusing  and unclear but that could be looked at in another PR

File: cleanrl/td3_continuous_action_jax.py [hunk 1]
```diff
@@ -263,28 +270,28 @@ def actor_loss(params):
                 [
                     (
                         jax.device_get(actions)[0]
-                        + np.random.normal(0, max_action * args.exploration_noise, size=envs.single_action_space.shape[0])
+                        + np.random.normal(0, max_action * args.exploration_noise, size=envs.single_action_space.shape)
```
Comment (vwxyzjn) [inline cleanrl/td3_continuous_action_jax.py:273]: Is this a problem?

**Review by vwxyzjn — COMMENTED** [review]
Thanks for the PR. I think the only thing is to wait #385 to be merged, remove the video logging stuff, and we can start running the RLops process.

File: cleanrl/td3_continuous_action_jax.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply cleanrl/td3_continuous_action_jax.py:273]: Yeah, I should have made a note, this would allow weirder action spaces

**Comment by sdpkjc** [comment]
👋Hi there,

I've noticed that this PR hasn't had any activity for several months and it now only requires some clean-up work and conflict resolution to complete. I would like to volunteer to take over and finish off this work if there's no objection. Your efforts on this so far have been much appreciated, and I look forward to the possibility of being able to contribute to its completion.

**Comment by vwxyzjn** [comment]
@sdpkjc feel free to. Thanks so much!

**Comment by sdpkjc** [comment]
pr-377
pr-377-time

**Comment by sdpkjc** [comment]
pr-377-jax
pr-377-time

**Comment by sdpkjc** [comment]
I've already run the RLops process and everything is performing as expected. If there are no other issues, I will proceed to merge this branch.

**Comment by vwxyzjn** [comment]
LGTM! Feel free to merge.

**Comment by pseudo-rnd-thoughts** [comment]
@sdpkjc Apologies for not finishing this and thanks for doing the rest of it

**Pull request merged**